hack/.golint_failures

```diff
@@ -33,6 +33,8 @@ pkg/apis/core/v1/helper
 pkg/apis/core/v1/helper/qos
 pkg/apis/core/v1/validation
 pkg/apis/core/validation
+pkg/apis/discovery/v1alpha1
```

Comment:
why are we adding adding all these exclusions?

Reply:
This is required because of the generated functions like `Convert_v1alpha1_EndpointSliceControllerConfiguration_To_config_EndpointSliceControllerConfiguration` that need to include underscores in their name to be referenced by other generated code. As far as I can tell, this config package is in `.golint_failures` for each controller. The other additions come from the parent API PR, which I think had similar rationale.